OpenRA.Mods.Common/Activities/Move/MoveWithinRange.cs

```diff
@@ -48,15 +46,13 @@ protected override IEnumerable<CPos> CandidateMovementCells(Actor self)
 			var maxCells = (maxRange.Length + 1023) / 1024;
 			var minCells = minRange.Length / 1024;
 
-			var outerSq = maxRange.LengthSquared;
-			var innerSq = minRange.LengthSquared;
-			var center = Target.CenterPosition;
+			return map.FindTilesInAnnulus(targetPosition, minCells, maxCells)
+				.Where(c => AtCorrectRange(map.CenterOfCell(c)));
+		}
 
-			return map.FindTilesInAnnulus(targetPosition, minCells + 1, maxCells).Where(c =>
-			{
-				var dxSq = (map.CenterOfCell(c) - center).HorizontalLengthSquared;
-				return dxSq >= innerSq && dxSq <= outerSq;
-			});
+		bool AtCorrectRange(WPos origin)
+		{
+			return Target.IsInRange(origin, maxRange) && !Target.IsInRange(origin, minRange);
```

Comment:
The rest of the targeting code doesn't make such an adjustment - so that would break everything again because we could end up too close.